content/docs/user-guide/checkpoints.md

```diff
@@ -0,0 +1,195 @@
+# Checkpoints
+
+ML checkpoints are an important part of deep learning because ML engineers like
+to save the model files at checkpoints during a training process and return back
+when metrics start diverging.
+
```

Comment:
Some ideas that can be used in the intro: "... saving ML model weights as checkpoints is one of the best practices in deep learning projects... DVC automate this best practice ... and help users not only to have the history of checkpoints but also keep track of all code and data changes corresponded to the checkpoints ... DVC introduces reproducibility to a messy deep learning process"

Reply:
Also, we should mention how checkpoints are implemented in DVC (a separate 1-2 paragraph section). We use Git commits, but these are not regular commits and branches, these are "ephemeral" commits and branches that not over pollute your branches' workspace. After you you are happy with some of the experiments or checkpoints you can "materialize" an "ephemeral" experiment branch.